backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java

```diff
@@ -100,6 +100,21 @@ public User registerAlpacaApiKey(Long id, String key) {
         return userToUpdate;
     }
 
+    /**
+     * A method to verify given credentials against existing user records
+     * @param credentials A User object with email and unencrypted password
+     * @return The full User object from the database if login is successful, null otherwise
+     */
+    @Override
+    public User login(User credentials) {
```

Comment:
My reasoning behind using the user type, as mentioned in my commit message, is if we want to allow a user to login with either email or username, it can easily be done this way without breaking changes to the endpoint. We would just have to add a check if email is empty then use username. Also, even if we're using a user type as input, the client doesn't need to pass a full user, simply a json object with email and password is enough.
e.g.
```json
{
    "username": "user",
    "password": "mypass"
}
```
@razine-bensari what do you think?

Reply:
Makes sense. 

> > Nice! I left some comments. Good job! Tested locally and worked.
> > Also, do you think we should add a better message when the email or password is invalid? Right not it only sends NOT_FOUND to the user. Maybe we could have an "Invalid Credentials" or "Invalid Email or password" instead of the "Not Found" message returned by the backend.
> 
> @DPigeon
> My reasoning behind a generic error message, as mentioned in my commit message, is for security and privacy purposes. If we return a "wrong password", then a malicious user would know that a user with the email provided exists. If you don't think this is a sufficient reason, I can change it.

"Wrong password" like you mentioned is very bad and is not an option I considered. But "Invalid Login" would be better in my opinion than "Not found". I feel like "Not found" does not makes sense. It does make sense in the backend but not from a user's perspective when loggin in.